spyder/plugins/maininterpreter/container.py

```diff
@@ -53,31 +53,28 @@ def update_actions(self):
 
     @on_conf_change(option=['default', 'custom_interpreter', 'custom'])
     def section_conf_update(self, option, value):
-        if option in ['default', 'custom_interpreter', 'custom'] and value:
-            self._update_status()
-            self.sig_interpreter_changed.emit()
-
-            # Set new interpreter
+        if ((option == 'default' and value) or
+                (option == 'custom' and not value)):
+            executable = get_python_executable()
+        else:
             executable = osp.normpath(self.get_conf('custom_interpreter'))
-            if (option in ['custom', 'custom_interpreter'] and
-                    osp.isfile(executable)):
-                self.sig_add_to_custom_interpreters_requested.emit(executable)
+            self.sig_add_to_custom_interpreters_requested.emit(executable)
+        # Setting executable option that will be used by other plugins in Spyder.
+        if executable != self.get_conf('executable'):
+                self.set_conf('executable', executable)
```

Comment:
Please indent correctly this line to 4 spaces (it seems it has 8 spaces of indentation).